sql-plugin/src/main/scala/org/apache/spark/sql/rapids/mathExpressions.scala

```diff
@@ -590,9 +573,19 @@ abstract class GpuRoundBase(child: Expression, scale: Expression) extends GpuBin
     val lhsValue = value.getBase
     val scaleVal = scale.getValue.asInstanceOf[Int]
 
-    dataType match {
-      case DecimalType.Fixed(_, scaleVal) =>
-        lhsValue.round(scaleVal, roundMode)
+    child.dataType match {
+      case DecimalType.Fixed(_, s) =>
+        // Only needs to perform round when required scale < input scale
+        val rounded = if (scaleVal < s) {
+          lhsValue.round(scaleVal, roundMode)
+        } else {
+          lhsValue.incRefCount()
+        }
+        withResource(rounded) { _ =>
+          // Fit the output datatype
+          rounded.castTo(
+            DecimalUtil.createCudfDecimal(dataType.asInstanceOf[DecimalType]))
```

Comment:
This cast seems unnecessary in most cases, may be worth checking if the desired cudf type is already the type computed to avoid an unnecessary copy.